src/recommendation/src/main/scala/RankingAdapter.scala

```diff
@@ -0,0 +1,382 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.recommendation._
+import org.apache.spark.ml.util._
+import org.apache.spark.ml.{Estimator, Model, Transformer}
+import org.apache.spark.sql._
+import org.apache.spark.sql.expressions.Window
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.functions.{collect_list, rank => r}
+import org.apache.spark.sql.types.{ArrayType, FloatType, IntegerType, StructType}
+
+import scala.collection.mutable
+import scala.util.Random
+
+trait RankingParams extends Params {
+  val minRatingsPerUser: IntParam = new IntParam(this,
+    "minRatingsPerUser",
+    "min ratings for users > 0",
+    ParamValidators.inRange(0, Integer.MAX_VALUE))
+
+  val minRatingsPerItem: IntParam = new IntParam(this,
+    "minRatingsPerItem",
+    "min ratings for items > 0",
+    ParamValidators.inRange(0, Integer.MAX_VALUE))
+
+  /** @group setParam */
+  def setMinRatingsPerUser(value: Int): this.type = set(minRatingsPerUser, value)
```

Comment:
small style: can you put the get/set together under the param, it is easier to read that way